Products/ZenUI3/browser/resources/js/zenoss/eventDetail.js

```diff
@@ -467,6 +467,7 @@ Ext.onReady(function() {
             this.addSection(eventDetailsSection);
 
             this.checkCustomizations();
+            Zenoss.Security.onPermissionsChange(this.updateEventActions, this);
```

Comment:
I don't understand what's accomplished by using `updateEventActions` as a callback when `Zenoss.Security.setContext` is called (the `onPermissionsChange` callback is called when `setContext` is called).

It seems like the `update` function would handle all the needed permission settings.

Reply:
setContext pulls permissions from the backend, I need to make sure I get permissions before I call updateEventActions